Framework/src/TaskRunnerFactory.cxx

```diff
@@ -28,13 +29,27 @@ TaskRunnerFactory::create(std::string taskName, std::string configurationSource,
   qcTask.setResetAfterPublish(resetAfterPublish);
 
   DataProcessorSpec newTask{
-    taskName,
+    qcTask.getDeviceName(),
     qcTask.getInputsSpecs(),
     Outputs{ qcTask.getOutputSpec() },
-    adaptFromTask<TaskRunner>(std::move(qcTask))
+    AlgorithmSpec{},
+    qcTask.getOptions()
   };
+  // this needs to be moved at the end
+  newTask.algorithm = adaptFromTask<TaskRunner>(std::move(qcTask));
 
   return std::move(newTask);
 }
 
+void TaskRunnerFactory::customizeInfrastructure(std::vector<framework::CompletionPolicy>& policies)
+{
+  auto matcher = [](framework::DeviceSpec const& device) {
+    return device.name.find(TaskRunner::createTaskRunnerIdString()) != std::string::npos;
```

Comment:
This is why it was necessary to change the name of the device in the DPL workflow. To apply customization to specific devices, we need to recognize them and unfortunately the only reasonable way was to match some string in their name.

Reply:
ok